build errors on testing pipelines due to check_library_exists

While testing the conda package, the windows build failed with 

    2020-08-07T12:17:32.5948967Z Call Stack (most recent call first):
    2020-08-07T12:17:32.5949504Z   CMakeLists.txt:644 (check_library_exists)
    2020-08-07T12:17:32.5950100Z This warning is for project developers.  Use -Wno-dev to suppress it.
    2020-08-07T12:17:32.5950557Z 
    2020-08-07T12:17:32.6012786Z CMake Error at C:/Miniconda/conda-bld/python-libsbml_1596802439467/work/build/temp.win-amd64-3.6/Release/CMakeFiles/CMakeTmp/CMakeLists.txt:15 (target_link_libraries):
    2020-08-07T12:17:32.6013865Z   Syntax error in cmake code at
    2020-08-07T12:17:32.6014174Z 
    2020-08-07T12:17:32.6014699Z     C:/Miniconda/conda-bld/python-libsbml_1596802439467/work/build/temp.win-amd64-3.6/Release/CMakeFiles/CMakeTmp/CMakeLists.txt:15
    2020-08-07T12:17:32.6015158Z 
    2020-08-07T12:17:32.6015482Z   when parsing string
    2020-08-07T12:17:32.6015751Z 
    2020-08-07T12:17:32.6016191Z     C:\Miniconda\conda-bld\python-libsbml_1596802439467\work\install_dependencies_win-amd64\lib\libbz2.lib
    2020-08-07T12:17:32.6016618Z 
    2020-08-07T12:17:32.6016934Z   Invalid character escape '\M'.
    2020-08-07T12:17:32.6017218Z 
    2020-08-07T12:17:32.6017473Z 
    2020-08-07T12:17:32.6030830Z CMake Error at C:/Miniconda/conda-bld/python-libsbml_1596802439467/_build_env/Library/share/cmake-3.18/Modules/CheckLibraryExists.cmake:67 (try_compile):
    2020-08-07T12:17:32.6031774Z   Failed to configure test project build system.
    2020-08-07T12:17:32.6032372Z Call Stack (most recent call first):
    2020-08-07T12:17:32.6032915Z   CMakeLists.txt:644 (check_library_exists)
    2020-08-07T12:17:32.6033335Z 

oddly, it stumbles over the windows path separator. So we need to transform it first.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -641,7 +641,8 @@ if(WITH_BZIP2)
   list(APPEND SWIG_EXTRA_ARGS -DUSE_BZ2)
 
     # make sure that we have a valid bzip2 library
-    check_library_exists("${LIBBZ_LIBRARY}" "BZ2_bzCompressInit" "" LIBBZ_FOUND_SYMBOL)
+    file(TO_CMAKE_PATH "${LIBBZ_LIBRARY}" LIBBZ2_CMAKE_PATH)
+    check_library_exists("${LIBBZ2_CMAKE_PATH}" "BZ2_bzCompressInit" "" LIBBZ_FOUND_SYMBOL)
     if(NOT LIBBZ_FOUND_SYMBOL)
         # this is odd, but on windows this check always fails! must be a
         # bug in the current cmake version so for now only issue this
@@ -742,7 +743,8 @@ if(WITH_ZLIB)
   list(APPEND SWIG_EXTRA_ARGS -DUSE_ZLIB)
 
     # make sure that we have a valid zip library
-    check_library_exists("${LIBZ_LIBRARY}" "gzopen" "" LIBZ_FOUND_SYMBOL)
+    file(TO_CMAKE_PATH "${LIBZ_LIBRARY}" LIBZ_CMAKE_PATH)
+    check_library_exists("${LIBZ_CMAKE_PATH}" "gzopen" "" LIBZ_FOUND_SYMBOL)
     if(NOT LIBZ_FOUND_SYMBOL)
         # this is odd, but on windows this check always fails! must be a
         # bug in the current cmake version so for now only issue this

